### ISSUE TITLE
Ignore function re-definition in comment for arg count lint

### ISSUE THREAD

**Issue by Akeboshiwind** [clj-kondo/clj-kondo#2020]
**version**

clj-kondo v2023.03.17

**platform**

I'm using the version installed by brew on a M2 Macbook Air.

**problem**

If you re-define a function in a comment, the linting seems to be based on that.
At least for the arg count lint.

**repro**

```clojure
(ns test)

(defn my-fn
  ([a b]
   (+ a b))
  ([a b c]
   (+ a b c)))

(comment
  (defn my-fn [a b]
    (+ a b)))

(my-fn 1 2 3)
```

```bash
$ cat test.clj | clj-kondo --lint -
<stdin>:13:1: error: test/my-fn is called with 3 args but expects 2
linting took 15ms, errors: 1, warnings: 0
```

If you move the comment above the function definition the error goes away.

**expected behavior**

I'm expecting clj-kondo to not use the function definition in the comment for code outside the comment.

**Comment by borkdude** [comment]
@Akeboshiwind 

This is a bit of an edge case where there doesn't exist one satisfactory answer, I think, since what should happen when you call `(my-fn 1 2 3)` inside the comment block? etc

You can ignore comment forms altogether with:

```
{:skip-comments true}
```

To tune linting behavior in comments you can also use:

```
{:config-in-comment ...}
```

**Comment by Akeboshiwind** [comment]
Seems like a reasonable solution to me 👍
